src/arch/helpers390x_128.h

```diff
@@ -0,0 +1,535 @@
+//          Copyright Naoki Shibata 2010 - 2019.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+#if CONFIG == 130 || CONFIG == 140 || CONFIG == 131 || CONFIG == 141
+
+#if !defined(__VX__)
+#error This helper is for IBM s390x.
+#endif
+
+#if (CONFIG == 140 || CONFIG == 141) && __ARCH__ < 12
+#error Please specify -march=z14.
+#endif
+
+#else
+#error CONFIG macro invalid or not defined
+#endif
+
+#define ENABLE_DP
+#define LOG2VECTLENDP 1
+#define VECTLENDP (1 << LOG2VECTLENDP)
+
+#define ENABLE_SP
+#define LOG2VECTLENSP (LOG2VECTLENDP+1)
+#define VECTLENSP (1 << LOG2VECTLENSP)
+
+#if CONFIG == 130 || CONFIG == 140
+#define ENABLE_FMA_DP
+#define ENABLE_FMA_SP
+#endif
+
+#define ACCURATE_SQRT
+#define FULL_FP_ROUNDING
+
+#include <vecintrin.h>
+
+#include <stdint.h>
+#include <math.h>
+#include "misc.h"
+
+#define VECTOR __attribute__((vector_size(16)))
```

Comment:
same here, why not using compiler token `__vector` instead?

Reply:
Done.